File: src/main/java/de/twometer/amongus/gui/GameEndPage.java
package de.twometer.amongus.gui;

import de.twometer.amongus.core.AmongUs;
import de.twometer.amongus.model.PlayerRole;

public class GameEndPage extends BasePage {

    public GameEndPage() {
        super("GameEnd.html");
        AmongUs.get().getScheduler().runLater(10000, () -> AmongUs.get().getGuiManager().showPage(new LobbyPage()));
    }

    @Override
    public void onDomReady() {
        super.onDomReady();
        var s = amongUs.getSession();
        var me = amongUs.getSession().getMyself();
        if (s.winners != me.role) {
            context.call("setDefeat");
            amongUs.getSoundFX().play("Defeat.ogg");
        } else {
            amongUs.getSoundFX().play("Victory.ogg");
        }

        for (var player : s.getPlayers()) {
            context.call("addPlayer", player.username, player.role == PlayerRole.Impostor);
        }
    }
}

File: src/main/kotlin/de/twometer/amogus/gui/GameEndPage.kt
package de.twometer.amogus.gui

import de.twometer.amogus.client.AmongUsClient
import de.twometer.amogus.client.StateManager
import de.twometer.amogus.model.GameState
import de.twometer.amogus.model.PlayerRole
import de.twometer.neko.audio.SoundEngine

class GameEndPage : BasePage("GameEnd.html") {

    override fun onLoaded() {
        val session = AmongUsClient.session ?: return
        val self = session.myself

        if (session.winners != self.role) {
            call("setDefeat")
            SoundEngine.play("Defeat.ogg")
        } else {
            SoundEngine.play("Victory.ogg")
        }

        session.players.forEach {
            call("addPlayer", it.username, it.role == PlayerRole.Impostor)
        }

        AmongUsClient.mainScheduler.runLater(10000) {
            session.taskProgress = 0f
            session.surveillanceActive = false
            PageManager.overwrite(LobbyPage())
            StateManager.changeGameState(GameState.Menus)
        }
    }

}